module/zfs/dmu_recv.c

```diff
@@ -1259,10 +1259,10 @@ receive_read(dmu_recv_cookie_t *drc, int len, void *buf)
 		    len - done, &resid);
 		if (resid == len - done) {
 			/*
-			 * Note: ECKSUM indicates that the receive
+			 * Note: ECKSUM or EPIPE indicates that the receive
 			 * was interrupted and can potentially be resumed.
 			 */
-			drc->drc_err = SET_ERROR(ECKSUM);
+			drc->drc_err = SET_ERROR(EPIPE);
```

Comment:
Generally I recommend that we add new errno's (to zfs_errno_t) rather than repurpose the standard ones.  But in this case it does seem like EPIPE is somewhat appropriate, since the description is `Broken pipe`.  However, I think the argument could still be made for a new errno, since you can get this error on non-pipes (e.g. `zfs receive <stream.trunc`)

Reply:
Also strictly speaking, EPIPE is for when a pipe you're writing to has hung up on you, it's not quite right. I tweaked it to have a custom zfs errno.